### PR TITLE
Integrating PEP8 compliance

### PR DESCRIPTION
**Pull request by rosesyrett**
Added class variable in stores/protocol to allow specifying error responses related directly to loading/unloading the Hkl object store, and removed 'errorDefinitions.py' file from src, instead moving it into errors/definitions.py

### CODE REVIEW

File: src/diffcalc_API/stores/protocol.py [hunk 1]
```diff
@@ -8,6 +8,8 @@ class HklCalcStore(Protocol):
     Protocol, or interface, for interacting with the Hkl object.
     """
 
+    responses: Dict[Union[int, str], Dict[str, Any]]
```
Comment (callumforrester) [inline src/diffcalc_API/stores/protocol.py:11]: I think these should be handled in the API layer. Given the work you've put in to separate the store from the API, you don't want to re-couple the two by putting API response information in the store.

Comment (rosesyrett) [reply src/diffcalc_API/stores/protocol.py:11]: The issue I've faced with this is, I've got two errors that are directly related to how I open/close the pickled file. Presumably, when I implement some sort of database, I might have more errors that I'll want to propagate based on this. So, the error responses here are purely related to how the store operates, therefore it makes sense to bundle the errors into the object which defines how the store is managed.

File: src/diffcalc_API/config.py [hunk 2]
```diff
@@ -1,11 +1,10 @@
 # this file is for defining constants
 
-savePicklesFolder = "/dls/tmp/ton99817/diffcalc_pickles"
-VectorProperties = ["n_hkl", "n_phi", "surf_nhkl", "surf_nphi"]
-pickledNames = {"ub", "constraints", "hkl"}
-constraintsWithNoValue = {"a_eq_b", "bin_eq_bout", "mu_is_gam", "bisect"}
+save_pickles_folder = "/dls/tmp/ton99817/diffcalc_pickles"
+vector_properties = ["n_hkl", "n_phi", "surf_nhkl", "surf_nphi"]
+constraints_with_no_value = {"a_eq_b", "bin_eq_bout", "mu_is_gam", "bisect"}
```
Comment (garryod) [inline src/diffcalc_API/config.py]: These are constants, therefore should be all caps, e.g. `SAVE_PICKLES_FOLDER`

File: src/diffcalc_API/errors/definitions.py [hunk 3]
```diff
@@ -39,45 +36,14 @@ def all_codes(self):
...
 #######################################################################################
 
 
-class codes(ErrorCodes):
-    attempting_to_overwrite = 405
-    check_file_exists = 404
-
-
-allResponses = {
+all_responses: Dict[Union[int, str], Dict[str, Any]] = {
```
Comment (garryod) [inline src/diffcalc_API/errors/definitions.py]: This is also a constant, so should be all caps

File: src/diffcalc_API/errors/hkl.py [hunk 4]
```diff
@@ -3,22 +3,26 @@
...
+    all_responses,
+)
 
 
-class codes(ErrorCodes):
+class Codes(ErrorCodes):
     check_valid_miller_indices = 400
     check_valid_scan_bounds = 400
-    calculate_UB_matrix = 400
+    calculate_ub_matrix = 400
```
Comment (garryod) [inline src/diffcalc_API/errors/hkl.py]: Enum keys are typically also all caps, owing to being constants

File: src/diffcalc_API/errors/ub.py [hunk 5]
```diff
@@ -0,0 +1,73 @@
...
+    all_responses,
+)
+from diffcalc_API.models.ub import SetLatticeParams
+
+
+class Codes(ErrorCodes):
+    check_params_not_empty = 400
+    get_reflection = 403
+    get_orientation = 403
+    check_property_is_valid = 400
```
Comment (garryod) [inline src/diffcalc_API/errors/ub.py]: Enum keys are typically also all caps, owing to being constants

File: src/diffcalc_API/errors/ub.py [hunk 6]
```diff
@@ -0,0 +1,73 @@
...
+    return orientation
+
+
+def check_property_is_valid(property: str) -> None:
+    if property not in vector_properties:
+        raise DiffcalcAPIException(
+            status_code=Codes.check_property_is_valid,
+            detail=f"invalid property. Choose one of: {vector_properties}",
+        )
+    return
```
Comment (garryod) [inline src/diffcalc_API/errors/ub.py]: This `return` can be omitted

File: src/diffcalc_API/services/constraints.py [hunk 7]
```diff
@@ -0,0 +1,60 @@
...
+
+    await store.save(name, hklcalc)
+
+
+async def set_constraint(
+    name: str,
+    property: str,
+    value: Union[float, bool],
+    store: HklCalcStore,
+):
```
Comment (garryod) [inline src/diffcalc_API/services/constraints.py]: This could do with an explicit return type

**Review by garryod — CHANGES_REQUESTED** [review]

File: src/diffcalc_API/stores/protocol.py (hunk 1 above)
Comment (callumforrester) [reply src/diffcalc_API/stores/protocol.py:11]: From discussion, we will merge this for now and investigate alternatives

**Review by callumforrester — APPROVED** [review]

**Review by garryod — APPROVED** [review]

**Pull request merged**